src/Compilers/CSharp/Portable/Symbols/Metadata/PE/MetadataDecoder.cs

```diff
@@ -127,8 +127,16 @@ protected override TypeSymbol LookupNestedTypeDefSymbol(TypeSymbol container, re
         {
             try
             {
-                AssemblySymbol assembly = moduleSymbol.GetReferencedAssemblySymbols()[referencedAssemblyIndex];
-                return assembly.LookupTopLevelMetadataType(ref emittedName, digThroughForwardedTypes: true);
+                var referencedAssemblies = moduleSymbol.GetReferencedAssemblySymbols();
+                if (referencedAssemblyIndex < referencedAssemblies.Length)
+                {
+                    AssemblySymbol assembly = referencedAssemblies[referencedAssemblyIndex];
+                    return assembly.LookupTopLevelMetadataType(ref emittedName, digThroughForwardedTypes: true);
+                }
+                else
+                {
+                    return new UnsupportedMetadataTypeSymbol();
```

Comment:
>return new UnsupportedMetadataTypeSymbol(); [](start = 20, length = 43)

Please add a comment about when do we expect to get here. Also, please add an assert or better throw if current assembly is not a core assembly.